VirtualFleetDocker.py

```diff
@@ -0,0 +1,351 @@
+    args = parse_arguments()
+    check_paths(args)
+
+    with open(args.config) as file:
+        return json.load(file)
+
+
+def stop_running_containers(image_names):
+    """
+    Stops and removes Docker containers based on the provided image names.
+    """
+    if not image_names:
+        return
+    docker_client = docker.from_env()
+    for container in docker_client.containers.list():
+        if container.image.tags and any(
+            image_name in container.image.tags[0] for image_name in image_names
+        ):
+            logging.info(
+                f"Stopping and removing container {container.image.tags[0]} with id {container.short_id}"
+            )
+            container.stop()
+            container.remove()
+    docker_client.close()
+
+
+def is_port_available(port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex(("localhost", port)) != 0
+
+
+def initialize_program(settings):
+    if settings["stop-running-containers"]:
+        stop_running_containers(
+            [
+                f"{settings['external-server-docker-image']}:",
+                f"{settings['gateway-docker-image']}:",
+                f"{settings['vehicle-docker-image']}:",
+                f"{settings['vernemq-docker-image']}:",
+            ]
+        )
+
+    check_vehicles_uniqueness(settings["vehicles"])
+    os.makedirs(os.path.abspath(TMP_CONFIG_DIR), mode=0o777, exist_ok=True)
+    for vehicle in settings["vehicles"]:
+        create_config_files(vehicle)
+
+
+def run_program(settings):
+    client = docker.from_env()
+    port = settings["start-port"]
+    logging.info("Starting docker containers")
+    start_mqtt_broker_container(client, settings)
+
+    for vehicle in settings["vehicles"]:
+        while not is_port_available(port):
+            logging.info(f"Port {port} is not available, trying next one")
+            port += 1
+
+        if not (1024 < port < 65535):
+            raise PortOutOfRangeException()
+
+        logging.info(f"Starting vehicle {vehicle['name']} with port {port}")
+        start_containers(client, settings, vehicle, port)
+        port += 1
+
+    logging.info("Start was successful")
+    end = False
+    while not end:
+        for container in running_containers:
+            container.reload()
+            if not container.attrs["State"]["Running"]:
+                container_name = container_id_name_dictionary[container.short_id]
+                logging.info(
+                    f"Error, container {container_name} with id {container.short_id} is not running, app will exit, container output: \n ==== \n{container.logs(tail=NUMBER_OF_LOG_LAST_LINES_TO_SHOW).decode()} \n ==== \n"
+                )
+                end = True
+        if not end:  # don't want to wait 5 seconds, if containers crash in the beginning
+            time.sleep(5)
+    stop_containers()
+    remove_tmp_config_files()
+
+
+def is_container_running(docker_client, image_name):
```

Comment:
functions named `is_*` should return bool. Rename to something like get_container_if_running